File: app/src/main/java/com/originb/inkwisenote2/modules/handwrittennotes/data/Stroke.java
package com.originb.inkwisenote2.modules.handwrittennotes.data;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Represents a single stroke in a handwritten note
 */
public class Stroke implements Serializable {
    private static final long serialVersionUID = 1L;
    
    private int color;
    private float width;
    private List<StrokePoint> points;
    
    public Stroke() {
        this.points = new ArrayList<>();
    }
    
    public Stroke(int color, float width) {
        this.color = color;
        this.width = width;
        this.points = new ArrayList<>();
    }
    
    public int getColor() {
        return color;
    }
    
    public void setColor(int color) {
        this.color = color;
    }
    
    public float getWidth() {
        return width;
    }
    
    public void setWidth(float width) {
        this.width = width;
    }
    
    public List<StrokePoint> getPoints() {
        return points;
    }
    
    public void setPoints(List<StrokePoint> points) {
        this.points = points;
    }
    
    public void addPoint(StrokePoint point) {
        if (points == null) {
            points = new ArrayList<>();
        }
        points.add(point);
    }
    
    public void addPoint(float x, float y, float pressure) {
        addPoint(new StrokePoint(x, y, pressure));
    }
    
    public void addPoint(float x, float y, float pressure, long timestamp) {
        addPoint(new StrokePoint(x, y, pressure, timestamp));
    }
} 

File: app/src/main/java/com/originb/inkwisenote2/modules/handwrittennotes/data/Stroke.kt
package com.originb.inkwisenote2.modules.handwrittennotes.data

import java.io.Serializable

/**
 * Represents a single stroke in a handwritten note
 */
class Stroke : Serializable {
    var color: Int = 0
    var width: Float = 0f
    var points: MutableList<StrokePoint?>?

    constructor() {
        this.points = ArrayList<StrokePoint?>()
    }

    constructor(color: Int, width: Float) {
        this.color = color
        this.width = width
        this.points = ArrayList<StrokePoint?>()
    }

    fun addPoint(point: StrokePoint?) {
        if (points == null) {
            points = ArrayList<StrokePoint?>()
        }
        points!!.add(point)
    }

    fun addPoint(x: Float, y: Float, pressure: Float) {
        addPoint(StrokePoint(x, y, pressure))
    }

    fun addPoint(x: Float, y: Float, pressure: Float, timestamp: Long) {
        addPoint(StrokePoint(x, y, pressure, timestamp))
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}